plugins/by-name/neoclip/default.nix

```diff
@@ -235,8 +235,16 @@ helpers.neovim-plugin.mkNeovimPlugin {
   };
 
   extraConfig = cfg: {
-    extraPlugins = mkIf (
-      isBool cfg.settings.enable_persistent_history && cfg.settings.enable_persistent_history
-    ) [ pkgs.vimPlugins.sqlite-lua ];
+    assertions = [
+      {
+        assertion =
+          (isBool cfg.settings.enable_persistent_history && cfg.settings.enable_persistent_history)
+          -> config.plugins.sqlite-lua.enable;
+        message = ''
+          Nixvim (plugins.neoclip): The persistent history sqlite storage backend needs `sqlite-lua` to function as intended.
+          You can enable it by setting `plugins.sqlite-lua.enable` to `true`.
+        '';
+      }
+    ];
```

Comment:
This is technically a breaking change, since we're asking users to enable another plugin to pass the assertion.

I wonder about a transition period where we set `plugins.sqlite-lua.enable` with a low-priority[^1] definition, then show a warning if the option's `highestPrio` matches our definition.

[^1]: an option's default or `mkOptionDefault` would use prio 1500, so maybe 1400 would be suitable for our purposes?

Reply:
So you want it to be in between mkOptionDefault and mkDefault, in case someone sets their own mkDefault config? Were you thinking of setting this in each of the modules that are transitioning or in the sqlite-lua module?

Reply:
The sqlite-lua module will declare the `enable` option with `default = false` (implicitly priority 1500).

Plugin modules that _require_ sqlite-lua will set `enable = mkOverride 1490 true`, which is equivalent to the current behavior of installing the package.

A per-plugin deprecation warning can then check `options.plugins.sqlite-lua.enable.highestPrio` (and the value too I guess).